### DISCUSSION TITLE
what do I do

### DISCUSSION THREAD

**Discussion by jmastermills** [FunkinCrew/Funkin#1572] [Compiling help]
```Its says this what do I do?  Compiling group: haxe
cl.exe -Iinclude -IC:/HaxeToolkit/haxe/lib/discord_rpc/git/linc/ -nologo /WX- /fp:precise -DHX_WINDOWS -GR -O2(optim-std) -Zi(debug) -FdC:\Users\cgjal\Downloads\Funkin-master (2)\Funkin-master\export\release\windows\obj\obj/msvc1964-nc/vc.pdb(debug) -Od(debug) -O2(release) -Os(optim-size) -FS -Oy- -c -EHs -GS- -IC:/HaxeToolkit/haxe/lib/hxcpp/4,2,1/include -DHXCPP_M64 -DHXCPP_VISIT_ALLOCS(haxe) -DHX_SMART_STRINGS(haxe) -DHXCPP_API_LEVEL=400(haxe) -D_CRT_SECURE_NO_DEPRECATE -D_ALLOW_MSC_VER_MISMATCH -D_ALLOW_ITERATOR_DEBUG_LEVEL_MISMATCH -DHX_WIN_MAIN(main) -wd4996 ... tags=[haxe,static]

src/flixel/input/actions/_FlxSteamController/FlxSteamControllerMetadata.cpp
src/sys/io/_Process/Stdout.cpp
src/openfl/display/_internal/Context3DElementType.cpp
Error: FlxSteamControllerMetadata.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\yvals.h(12): fatal error C1083: Cannot open include file: 'crtdbg.h': No such file or directory
Error: Context3DElementType.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\yvals.h(12): fatal error C1083: Cannot open include file: 'crtdbg.h': No such file or directory```

**Comment by libauror-a** [comment]
What command did you run?

**Comment by jmastermills** [comment]
lime test windows

**Comment by bonkmaykr** [comment]
Hello, please re-paste this by surrounding them with triple-backticks ``` so that Github's markdown parser doesn't make it harder to read.

**Comment by ghost** [comment]
> Hello, please re-paste this by surrounding them with triple-backticks ``` so that Github's markdown parser doesn't make it harder to read.

```
Compiling group: haxe
cl.exe -Iinclude -IC:/HaxeToolkit/haxe/lib/discord_rpc/git/linc/ -nologo /WX- /fp:precise -DHX_WINDOWS -GR -O2(optim-std) -Zi(debug) -FdC:\Users\cgjal\Downloads\Funkin-master (2)\Funkin-master\export\release\windows\obj\obj/msvc1964-nc/vc.pdb(debug) -Od(debug) -O2(release) -Os(optim-size) -FS -Oy- -c -EHs -GS- -IC:/HaxeToolkit/haxe/lib/hxcpp/4,2,1/include -DHXCPP_M64 -DHXCPP_VISIT_ALLOCS(haxe) -DHX_SMART_STRINGS(haxe) -DHXCPP_API_LEVEL=400(haxe) -D_CRT_SECURE_NO_DEPRECATE -D_ALLOW_MSC_VER_MISMATCH -D_ALLOW_ITERATOR_DEBUG_LEVEL_MISMATCH -DHX_WIN_MAIN(main) -wd4996 ... tags=[haxe,static]

src/flixel/input/actions/_FlxSteamController/FlxSteamControllerMetadata.cpp
src/sys/io/_Process/Stdout.cpp
src/openfl/display/_internal/Context3DElementType.cpp
Error: FlxSteamControllerMetadata.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\yvals.h(12): fatal error C1083: Cannot open include file: 'crtdbg.h': No such file or directory
Error: Context3DElementType.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\yvals.h(12): fatal error C1083: Cannot open include file: 'crtdbg.h': No such file or directory
```
There.

**Comment by ghost** [comment]
> lime test windows
> 
> 

have you tried installing the 4gb of crap

**Comment by jmastermills** [comment]
no

**Comment by isakube** [comment]
do it

**Comment by jmastermills** [comment]
how do I download it at

**Comment by bonkmaykr** [comment]
> > Hello, please re-paste this by surrounding them with triple-backticks ``` so that Github's markdown parser doesn't make it harder to read.
> 
> ```
> Compiling group: haxe
> cl.exe -Iinclude -IC:/HaxeToolkit/haxe/lib/discord_rpc/git/linc/ -nologo /WX- /fp:precise -DHX_WINDOWS -GR -O2(optim-std) -Zi(debug) -FdC:\Users\cgjal\Downloads\Funkin-master (2)\Funkin-master\export\release\windows\obj\obj/msvc1964-nc/vc.pdb(debug) -Od(debug) -O2(release) -Os(optim-size) -FS -Oy- -c -EHs -GS- -IC:/HaxeToolkit/haxe/lib/hxcpp/4,2,1/include -DHXCPP_M64 -DHXCPP_VISIT_ALLOCS(haxe) -DHX_SMART_STRINGS(haxe) -DHXCPP_API_LEVEL=400(haxe) -D_CRT_SECURE_NO_DEPRECATE -D_ALLOW_MSC_VER_MISMATCH -D_ALLOW_ITERATOR_DEBUG_LEVEL_MISMATCH -DHX_WIN_MAIN(main) -wd4996 ... tags=[haxe,static]
> 
> src/flixel/input/actions/_FlxSteamController/FlxSteamControllerMetadata.cpp
> src/sys/io/_Process/Stdout.cpp
> src/openfl/display/_internal/Context3DElementType.cpp
> Error: FlxSteamControllerMetadata.cpp
> C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\yvals.h(12): fatal error C1083: Cannot open include file: 'crtdbg.h': No such file or directory
> Error: Context3DElementType.cpp
> C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.28.29910\include\yvals.h(12): fatal error C1083: Cannot open include file: 'crtdbg.h': No such file or directory
> ```
> 
> There.

You're either missing dependencies, made a typo in your #includes, or deleted something on accident etc.
that "crtdbg.h" is a header file containing code needed by "yvals.h".

**Comment by suprstarrd** [comment]
From a modified README:

> To run it from your desktop (Windows, Mac, Linux) it can be a bit more involved. For Linux, you only need to open a terminal in the project directory and run 'lime test linux -debug' and then run the executable file in export/release/linux/bin. For Windows, you need to install Visual Studio Community 2019. While installing VSC, don't click on any of the options to install workloads. Instead, go to the individual components tab and choose the following:
> * MSVC v142 - VS 2019 C++ x64/x86 build tools (Latest)
> * Windows 10 SDK (10.0.19041.0)
> 
> This will install about 4GB of crap, but once that is done you can open up a command line in the project's directory and run `lime test windows -debug`. Once that command finishes (it takes forever even on a higher end PC), you can run FNF from the .exe file under export\release\windows\bin
> As for Mac, 'lime test mac -debug' should work, if not the internet surely has a guide on how to compile Haxe stuff for Mac.

**Comment by imasqueakydoor** [comment]
Make sure you dont lose

**Comment by Adamusique** [comment]
i have this error source/Character.hx:351: characters 10-16 : Warning : Potential typo detected (expected similar values are characters.AnimType.frames)
Sys_error("export/release/windows/obj/src/openfl/display3D/_Context3DVertexBufferFormat/Context3DVertexBufferFormat_Impl_.cpp: No such file or directory")
